src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala

```diff
@@ -137,7 +137,53 @@ trait RawlsGroupComponent {
         
         nextLevelUp.result.flatMap { nextGroups =>
           val nextCumulativeGroups = cumulativeGroups ++ nextGroups
-          listParentGroupsRecursive(nextGroups.toSet -- nextCumulativeGroups, nextCumulativeGroups)
+          listParentGroupsRecursive(nextGroups.toSet -- cumulativeGroups, nextCumulativeGroups)
```

Comment:
The logic looks good.  No need to add tests for this PR.  I added a story: https://broadinstitute.atlassian.net/browse/GAWB-419